packages/core/admin/admin/src/content-manager/components/RelationInput/RelationInput.js

```diff
@@ -1,31 +1,48 @@
+import React, { useRef, useState } from 'react';
 import PropTypes from 'prop-types';
-import React from 'react';
-import styled from 'styled-components';
+import styled, { keyframes } from 'styled-components';
+import { FixedSizeList as List } from 'react-window';
 
 import { ReactSelect } from '@strapi/helper-plugin';
 import { Badge } from '@strapi/design-system/Badge';
 import { Box } from '@strapi/design-system/Box';
-import { BaseLink } from '@strapi/design-system/BaseLink';
+import { Link } from '@strapi/design-system/Link';
 import { Icon } from '@strapi/design-system/Icon';
 import { FieldLabel, FieldError, FieldHint, Field } from '@strapi/design-system/Field';
 import { TextButton } from '@strapi/design-system/TextButton';
 import { Typography } from '@strapi/design-system/Typography';
-import { Loader } from '@strapi/design-system/Loader';
+// import { Loader } from '@strapi/design-system/Loader';
 
 import Cross from '@strapi/icons/Cross';
 import Refresh from '@strapi/icons/Refresh';
+import Loader from '@strapi/icons/Loader';
 
 import { Relation } from './components/Relation';
 import { RelationItem } from './components/RelationItem';
 import { RelationList } from './components/RelationList';
 import { Option } from './components/Option';
 
-const RelationItemCenterChildren = styled(RelationItem)`
-  div {
-    justify-content: center;
+const LinkEllipsis = styled(Link)`
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: inherit;
+`;
+
+const rotation = keyframes`
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(359deg);
   }
 `;
 
+// Temporary component - to replace with loading prop on TextButton after DS release
```

Comment:
Could you please prefix the comment with `TODO` so that we can won't forget it at the end?